swap_meet/vendor.py

```diff
@@ -1,2 +1,65 @@
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        try: 
+            self.inventory.remove(item)
+            return item
+        except ValueError:
+            return False
+        #  ValueError will be raised if item not in self.inventory
+
+    def get_by_category(self, category):
+        return [item for item in self.inventory if item.category == category]
+
+    def swap_items(self, other, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other.inventory:
+            return False    
+
+        self.add(their_item)
+        other.add(my_item)
+        self.remove(my_item)
+        other.remove(their_item)
+        return True
+    
+    def swap_first_item(self, other):
+        if not self.inventory or not other.inventory:
+            return False
+
+        return self.swap_items(other, self.inventory[0], other.inventory[0])
+
+    def get_best_by_category(self, category):
+        items_in_category = self.get_by_category(category)
+        
+        try:
+            return max(items_in_category, key=lambda item: item.condition)
+        except ValueError:
+            #  ValueError will be raised if items_in_category is empty
+            return None
```

Comment:
Nice use of max with a key function to help find the item with the "best" condition. We've seen enough python now, that you could try writing your own version of max, including an optional key function parameter. Give it a shot!

And nice error-handling approach since max can raise `ValueError`.